src/basic/mount-util.c

```diff
@@ -534,15 +534,20 @@ int repeat_unmount(const char *path, int flags) {
 }
 
 const char* mode_to_inaccessible_node(mode_t mode) {
+        struct stat target;
         switch(mode & S_IFMT) {
                 case S_IFREG:
                         return "/run/systemd/inaccessible/reg";
                 case S_IFDIR:
                         return "/run/systemd/inaccessible/dir";
                 case S_IFCHR:
-                        return "/run/systemd/inaccessible/chr";
+                        if (stat("/run/systemd/inaccessible/chr", &target))
```

Comment:
if you want to check if a path exists  please use access(…, F_OK) instead of stat().